xgcm/grid.py

```diff
@@ -645,7 +662,7 @@ def diff(
             boundary_discontinuity,
             vector_partner,
             keep_coords=keep_coords,
-        )
+        )*self.direction_sign
```

Comment:
I disagree with this, TBH. If not set explicitly in the call to `Grid(...)` nothing changes here. This would only be a slight 'black-box' situation if we detect directionality from CF-attributes (which I think is not *yet* done here). In that case we should just raise a proper warning IMO.

Implementing this on the lowest level makes sure it 'propagates' up to whatever more complex functions we come up with. This might be personal preference, but I would find it way more confusing (and it would require tons of doc changes) if `grid.diff(da, axis) / da_metric != grid.derivative(da,axis)`